### ISSUE TITLE
[Edge] Issues with Caret Browsing mode

### ISSUE THREAD

**Issue by Mgsy** [ckeditor/ckeditor5#738]
## 🐞 Is this a bug report or feature request? (choose one)

- Bug report

## 💻 Version of CKEditor

1.0.0-alpha2

## 📋 Steps to reproduce

1. Go to http://localhost:8125/ckeditor5-core/tests/manual/article.html.
2. Turn on the `Caret Browsing` mode with <kbd>F7</kbd>.
3. Put the caret at the end of the block.
4. Press <kbd>Arrow right</kbd>.

## ✅ Expected result

The selection moves to the next line.

## ❎ Actual result

Empty space has been inserted.

## 📃 Other details that might be useful

**GIF**
![bug_cke5](https://user-images.githubusercontent.com/20988892/34206836-6cf78884-e588-11e7-90da-097f91767746.gif)

**Other information**
This bug is also reproducible in https://ckeditor5.github.io

OS: Windows 10
Browser: Edge 16

**Comment by Mgsy** [comment]
Ok, I've figured out where is the source of problem. It's all about `Caret Browsing` option which allows you to put the selection in any text on the page and i.e. select it without using the mouse. You can enable it by pressing `F7`. 

Empty space will insert only when `Caret Browsing` mode is enabled.

**Comment by Mgsy** [comment]
Here is another case:

## 📋 Steps to reproduce

1. Go to http://localhost:8125/ckeditor5-core/tests/manual/article.html.
2. Turn on the `Caret Browsing` mode with <kbd>F7</kbd>.
3. Click on the blank space next to the side image's right border.
4. Double click on the same spot.
5. Type something.

## ❎ Actual result

Text has appeared inside the widget.

## 📃 Other details that might be useful

**GIF**
![bug_cke5](https://user-images.githubusercontent.com/20988892/34292144-de2a2aa2-e6ff-11e7-800d-d9bdaed19364.gif)

**Comment by Mgsy** [comment]
Moved from https://github.com/ckeditor/ckeditor5/issues/735.

## 📋 Steps to reproduce

1. Go to http://localhost:8125/ckeditor5-core/tests/manual/article.html.
2. Turn on the `Caret Browsing` mode with <kbd>F7</kbd>.
3. Select two block elements.
4. Press <kbd>Alt</kbd> + <kbd>F10</kbd>.
5. Press <kbd>Enter</kbd>.

## ✅ Expected result

Dropdown should reveal.

## ❎ Actual result

Selected elements disappear and the editor crashes.

## 📃 Other details that might be useful

**GIF**

![bug_cke5](https://user-images.githubusercontent.com/20988892/34160225-62943570-e4cc-11e7-95a3-4563b77a0013.gif)

**Error**
```
 SCRIPT5007: SCRIPT5007: Unable to get property 'parent' of undefined or null reference
```

**Comment by Reinmar** [comment]
Wow... I didn't know about the caret browsing mode. Is it on by default? Can you find more resources about it?

**Comment by Mgsy** [comment]
> Wow... I didn't know about the caret browsing mode. Is it on by default? Can you find more resources about it?

The caret browsing mode is an accessibility feature which is available in Edge, IE and Firefox. It's off by default. In Edge and IE you can simply activate it by pressing <kbd>F7</kbd> (it won't turn off after closing the browser), in Firefox you have to go to `Options -> General ` and check the `Always use the cursor keys to navigate within pages` input.

There's not much information about this feature. Basically, it's all about selecting text on the website without using the mouse. It allows you to put the selection in any text and use arrow keys, <kbd>End</kbd>, <kbd>Home</kbd>, <kbd>Shift</kbd> and <kbd>Ctrl</kbd> to select it. 

I've checked how the editor behaves while the caret browsing mode is enabled in Firefox and it works without any problems. If you want to use this feature in Chrome, you can install an extension (i.e. [Caret](http://thomaswilburn.net/caret/)) and it also doesn't have any negative impact on the editor. 

If you're curious how does the sample implementation looks like, check out the [Caret's GitHub repository](https://github.com/thomaswilburn/Caret).

**Comment by FilipTokarski** [comment]
I'm not able to reproduce those issues on Edge, apart from https://github.com/ckeditor/ckeditor5/issues/738#issuecomment-353568567 (but the editor does not crash).
There is hovewer a different behaviour: clicking `arrow right` does not get you to the next line.

Good news is that all those bugs are gone in the new upcoming Edge v.79.0.309.56

**Comment by Mgsy** [comment]
Fixed in the new Edge release.
